app/controllers/tasks_controller.rb

```diff
@@ -31,12 +35,40 @@ def create
     end
   end
 
+  def edit
+    @task 
+  end
+
+  def update
+    if @task.update(task_params)
+      redirect_to tasks_path, notice: 'Task was successfully updated.'
+    else
+      render :edit
+    end
+  end
+
+  def toggle_status
+    @task.update(status: @task.status == 1 ? 0 : 1)
+    respond_to do |format|
+      format.html { redirect_to tasks_path } 
+      format.js  
+    end
+  end
+
   private
 
   def task_params
     params.require(:task).permit(:title, :description, :status, :time_start, :time_end, :priority, :tag_list)
   end
 
+  def set_task
+    @task = Task.find(params[:id])
+  end
+
+  def set_tags
+    @tags = Tag.all 
```

Comment:
I saw you load current tasks in index, why do you need load all tags in system here? 
if in system had 1.000.000 tags records? what do you think about that when you load all records here?

Reply:
done